### PR TITLE
#1931 [feat]add dropdown menu and role, linkedInPortfolio and websitePortfo…

### PR DESCRIPTION
**Pull request by anne-creator**
[feat]add `dropdownMenu` and `role`, `linkedInPortfolio` and `websitePortfolio` inputfield to UserLandingPage.js

For the dropdown menu, it is currently working and also updating the Person data while selecting.
However, as described in the video, I didn't create a dropdown component under the UserProfile team that takes an array as the input. This is because the dropdown menu also has to update the Person data using the `onSelect={() => onCountryChange(item.name)}`.

What I would recommend is to copy all dropdownMenu components including ```
DropdownMenu,
  DropdownMenuTrigger,
  DropdownMenuContent,
  DropdownMenuItem
```, to our user profile project and then use it inside the `UserLandingPage`

I will clean up the code once we finalized and pr again

### CODE REVIEW

File: apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js [hunk 1]
```diff
@@ -11,6 +11,14 @@ import CheckboxField from './../../../common/Forms/Checkbox';
 import UploadProfilePicture from './../../../common/UploadProfilePicture';
 import Breadcrumb from './../../../../images/Onboarding/breadcrumb-frame.png';
 import Loader from './../../../common/Loader';
+// import {DropdownMenu} from "./../../../common/Dropdown/Dropdown";
```
Comment (chungthuang) [inline apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js:14]: Should we delete unused import?

File: apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js [hunk 2]
```diff
@@ -50,6 +59,7 @@ export default function UserLandingPage() {
   const [formValidation, setFormValidation] = useState(
     validate(initialFormValue)
   );
+  console.log(person);
```
Comment (chungthuang) [inline apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js:62]: Should we remove debug logs before merging?

**Comment by anne-creator** [comment]
Currently isolated DropDownMenu component (user-profile version) from UserLandingPage. 
Needs to add styled country label into the DropDownMenu component.

File: apps/user-profile/src/components/common/Dropdown/Dropdown.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuContent,
+  DropdownMenuItem,
+} from '@devlaunchers/components/components/DropdownMenu';
+function DropDownMenu({ menuItems = [], country, onCountryChange }) {
```
Comment (judeamos) [inline apps/user-profile/src/components/common/Dropdown/Dropdown.js:8]: `({ menuItems = [], **country**, **onCountryChange** })` those last two parameters should not be coupled to the "country" solution.  Remember this is a component that will be used throughout the the user profile project for different tasks.

File: apps/user-profile/src/components/common/Dropdown/Dropdown.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  return (
+    <DropdownMenu>
+      <p className="text-xs leading-4 block w-full box-border font-normal pd-3">
+        Country
+      </p>
+      <DropdownMenuTrigger asChild size="100%">
+        <div className="group">{country || 'Select...'}</div>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent size="large">
+        {menuItems.map((item) => (
```
Comment (judeamos) [inline apps/user-profile/src/components/common/Dropdown/Dropdown.js:18]: What if "item" is null, undefined, or an empty string?

File: apps/user-profile/src/components/common/Dropdown/Dropdown.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+          <DropdownMenuItem
+            key={item.code}
+            onSelect={() => onCountryChange(item.name)}
+          >
+            <button aria-label="fake-anchor">{item.name}</button>
+          </DropdownMenuItem>
+        ))}
```
Comment (judeamos) [inline apps/user-profile/src/components/common/Dropdown/Dropdown.js:25]: What if `item` is null, undefined, or an empty string?

File: apps/user-profile/src/components/common/Dropdown/Dropdown.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        Country
+      </p>
+      <DropdownMenuTrigger asChild size="100%">
+        <div className="group">{country || 'Select...'}</div>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent size="large">
+        {menuItems.map((item) => (
+          <DropdownMenuItem
+            key={item.code}
+            onSelect={() => onCountryChange(item.name)}
```
Comment (judeamos) [inline apps/user-profile/src/components/common/Dropdown/Dropdown.js:21]: Similar thing here, this is coupled to the country solution. What if someone wants to use this component in other places where `code` is not available.

File: apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js [hunk 7]
```diff
@@ -133,16 +138,20 @@ export default function UserLandingPage() {
...
 
-  function onHeadLineChange(e) {
-    setPerson({ ...person, headline: e.target.value });
+  function onLinkedInPortfolioChange(e) {
+    setPerson({ ...person, linkedInPortfolio: e.target.value });
+  }
+
+  function onWebsitePortfolioChange(e) {
+    setPerson({ ...person, websitePortfolio: e.target.value });
   }
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js:155]: Can you comment these out, don't delete it, just comment it out. This won't be going into the November release until the backend has been sorted out.

File: apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js [hunk 8]
```diff
@@ -283,10 +291,16 @@ export default function UserLandingPage() {
...
+              label="linkedInPortfolio (optional)"
+              onChange={onLinkedInPortfolioChange}
+              placeholder="www.linkedin.com"
+            />
+            <InputField
+              error=""
+              label="websitePortfolio (optional)"
+              onChange={onWebsitePortfolioChange}
+              placeholder="www.portfolio.com"
+            />
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js:303]: Also comment this out as well

File: apps/user-profile/src/content/countryData.json
Comment (judeamos) [inline apps/user-profile/src/content/countryData.json:1]: Awesome job 👍🏾

**Pull request merged**